tests/testsuite/cargo_add/change_rename_target/out/Cargo.toml

```diff
@@ -6,3 +6,6 @@ version = "0.0.0"
 
 [dependencies]
 some-package = { package = "my-package2", version = "99999.0.0", optional = true }
+
+[features]
+some-package = ["dep:some-package"]
```

Comment:
Can you double check that we have an `overwrite` test that will make sure we don't add `some-package` if `dep:some-package` is already in use elsewhere?

If so, I'd recommend making the test its own commit and make it the first in this PR and it should pass.  Then the behavior-change commit wouldn't change the test as nothing would be affected, making it clear to reviewers that it worked.

Reply:
It seems no `overwrite` test for that. I added one in this PR：tests/testsuite/cargo_add/overwrite_optional_with_optional